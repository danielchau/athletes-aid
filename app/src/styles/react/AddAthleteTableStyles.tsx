import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Add Athletes Table.
 */
export const addAthleteTableStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            padding: "8px",
            height: "100%",
            display: "flex",
            justifyContent: "center"
        },
        tableContainer: {
            width: "100%",
            height: "100%",
            padding: "8px",
            backgroundColor: "rgba(0, 33, 69, 0.05)",
            overflow: "scroll"
        },
        table: {
            width: "100%"
        },
        tableBodyContainer: {
            width: "2000px",
            overflow: "auto",
            height: "100%"
        },
        tableBody: {
            tableLayout: "fixed"
        },
        tableRow: {
            cursor: "pointer"
        },
        tooltip: {
            backgroundColor: "#fff",
            color: "#4f4f4f",
            boxShadow: theme.shadows[1]
        }
    })
);
