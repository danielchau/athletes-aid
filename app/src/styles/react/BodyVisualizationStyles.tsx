import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Body Visualization.
 */
export const bodyVisualizationStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            height: "100%"
        },
        imageContainer: {
            position: "relative",
            alignSelf: "center"
        },
        image: {
            width: "100%"
        },
        dot: {
            height: "15px",
            width: "15px",
            position: "absolute",
            borderRadius: "15px"
        },
        legendContainer: {
            borderLeft: "#dedede solid 1px"
        },
        legendItem: {
            display: "flex",
            color: "rgba(0, 0, 0, 0.54)",
            marginBottom: "2px"
        },
        dotLegend: {
            height: "10px",
            width: "10px",
            borderRadius: "10px",
            marginTop: "6px",
            marginRight: "4px"
        }
    })
);
