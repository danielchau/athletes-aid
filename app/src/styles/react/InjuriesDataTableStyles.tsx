import { makeStyles, Theme, createStyles, lighten } from "@material-ui/core/styles";

/**
 * React styles for the Injury Data Table.
 */
export const injuriesDataTableStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            marginTop: theme.spacing(3)
        },
        paper: {
            width: "100%",
            marginBottom: theme.spacing(2),
            backgroundColor: "rgba(0, 33, 69, 0.05)"
        },
        table: {
            minWidth: 750
        },
        tableWrapper: {
            overflowX: "auto"
        },
        tableRow: {
            cursor: "pointer"
        },
        visuallyHidden: {
            border: 0,
            clip: "rect(0 0 0 0)",
            height: 1,
            margin: -1,
            overflow: "hidden",
            padding: 0,
            position: "absolute",
            top: 20,
            width: 1
        }
    })
);

/**
 * React styles for the Injury Data Table Toolbar Header.
 */
export const injuriesDataTableToolbarStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            paddingLeft: theme.spacing(2),
            paddingRight: theme.spacing(1)
        },
        highlight:
            theme.palette.type === "light"
                ? {
                      color: theme.palette.primary.main,
                      backgroundColor: lighten(theme.palette.primary.light, 0.85)
                  }
                : {
                      color: theme.palette.text.primary,
                      backgroundColor: theme.palette.secondary.dark
                  },
        title: {
            flex: "1 1 100%"
        },
        export: {
            margin: "auto",
            color: theme.palette.secondary.main,
            display: "flex"
        }
    })
);
