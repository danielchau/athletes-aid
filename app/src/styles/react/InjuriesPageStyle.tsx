import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Injuries Page.
 */
export const injuriesPageStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            padding: "16px",
            backgroundColor: "#fff"
        },
        drawerOpen: {
            width: "calc(100vw - 240px)",
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.enteringScreen
            }),
            [theme.breakpoints.down("xs")]: {
                width: "100vw"
            }
        },
        drawerClosed: {
            width: "calc(100vw - 72px)",
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen
            }),
            [theme.breakpoints.down("xs")]: {
                width: "100vw"
            }
        },
        grid: {
            backgroundColor: "#fff"
        },
        paper: {
            padding: theme.spacing(2),
            textAlign: "center",
            color: theme.palette.text.secondary,
            backgroundColor: "rgba(0, 33, 69, 0.05)",
            height: "100%"
        },
        vizPaper: {
            padding: theme.spacing(2),
            height: "100%"
        },
        primaryStatisticContainer: {
            display: "flex",
            flexDirection: "column"
        },
        primaryStatisticValue: {
            overflow: "hidden",
            textOverflow: "ellipsis",
            fontSize: "calc((100vw - 240px)/16)"
        },
        primaryStatisticLabel: {
            paddingTop: "16px"
        },
        dateTimeContainer: {},
        textField: {
            marginLeft: theme.spacing(1),
            marginRight: theme.spacing(1),
            width: 200
        },
        dateField: {
            width: 200,
            marginRight: theme.spacing(1),
            marginTop: 0,
            marginBottom: 0
        },
        dateTimeButton: {
            marginLeft: "8px",
            marginTop: "6px"
        }
    })
);
