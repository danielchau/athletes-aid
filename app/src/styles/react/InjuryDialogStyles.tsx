import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryDialogStyles = makeStyles((theme: Theme) =>
    createStyles({
        appBar: {
            position: "relative"
        },
        title: {
            marginLeft: theme.spacing(2),
            flex: 1
        },
        dialogTitle: {
            padding: "0"
        },
        dialogPaper: {
            height: "100%",
            maxWidth: "600px",
            width: "50%",
            minWidth: "200px",
            position: "fixed",
            bottom: "0",
            right: "0",
            margin: "0"
        },
        dialogContentContainer: {
            padding: "0"
        },
        dialogContent: {
            padding: "16px"
        },
        dialogContentPaper: {
            backgroundColor: "#fafafa",
            padding: "8px",
            marginBottom: "8px"
        },
        notePaper: {
            backgroundColor: "#ffffff",
            padding: "8px",
            marginBottom: "8px"
        },
        noteContainer: {
            display: "flex",
            flexDirection: "row"
        },
        noteIcon: {
            margin: "auto",
            marginLeft: "8px",
            marginRight: "16px"
        },
        newNoteContainer: {
            display: "flex",
            flexDirection: "row",
            width: "100%",
            justifyContent: "center",
            alignItems: "center",
            alignSelf: "center"
        },
        newNote: {
            flexGrow: 1,
            paddingRight: "8px"
        },
        sendButton: {
            color: "#F2A71E"
        },
        activeSwitchContainer: {
            display: "flex",
            flexDirection: "row",
            justifyContent: "center",
            alignContent: "center",
            alignSelf: "center",
            paddingBottom: "8px"
        }
    })
);
