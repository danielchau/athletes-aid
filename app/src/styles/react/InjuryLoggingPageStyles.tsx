import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryLoggingPageStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            padding: "8px"
        },
        backButton: {
            marginRight: theme.spacing(1)
        },
        completedContainer: {
            height: "calc(100% - 116px)",
            textAlign: "center",
            paddingTop: "20%"
        },
        instructions: {
            marginTop: theme.spacing(1),
            marginBottom: theme.spacing(1)
        },
        paper: {
            color: theme.palette.text.secondary,
            backgroundColor: "#fafafa"
        },
        paperContent: {
            height: "100%",
            padding: "16px",
            paddingRight: "8px",
            overflow: "auto"
        },
        stepper: {
            backgroundColor: "#fafafa",
            borderRadius: "4px"
        },
        loggingContent: {
            height: "calc(100% - 168px)",
            marginBottom: "52px",
            marginTop: "8px",
            overflow: "hidden"
        },
        loggingBottomButtons: {
            position: "fixed",
            bottom: 0,
            right: 0,
            height: "52px",
            padding: "8px"
        },
        multilineInput: {
            width: "100%"
        }
    })
);
