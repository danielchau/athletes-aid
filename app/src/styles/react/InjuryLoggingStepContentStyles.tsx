import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryLoggingStepContentStyles = makeStyles((theme: Theme) =>
    createStyles({
        formControl: {
            minWidth: 200,
            width: "50%",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        textInput: {
            minWidth: 200,
            width: "100%",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        multilineInput: {
            width: "100%",
            paddingBottom: "16px",
            margin: "0px"
        },
        dateInput: {
            minWidth: 200,
            width: "50%",
            margin: "0px",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        checkboxInput: {
            minWidth: 200,
            width: "50%",
            margin: "0px",
            paddingRight: "8px",
            paddingBottom: "23px",
            paddingTop: "7px"
        },
        instructions: {
            textAlign: "center",
            marginBottom: theme.spacing(3),
            color: theme.palette.text.secondary,
            width: "100%"
        }
    })
);
