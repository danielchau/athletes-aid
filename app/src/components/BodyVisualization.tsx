import React from "react";
import { Injury } from "../util/types";
import { bodyVisualizationStyles } from "../styles/react/BodyVisualizationStyles";
// @ts-ignore
import Body from "../util/body.jpg";
import { Grid, Typography, Divider } from "@material-ui/core";

interface BodyVisualizationProps {
    injuries: Injury[];
}

export default function BodyVisualization(props: BodyVisualizationProps) {
    const classes = bodyVisualizationStyles({});

    const getStyles = (injury: Injury | string, isLegend: boolean) => {
        let color = "";
        let left = 0;
        let top = 0;

        let key = typeof injury == "string" ? injury : injury.locationOnBody;

        switch (key) {
            case "Abdominals":
                color = "#FE4365";
                top = 35;
                left = 24;
                break;
            case "Ankle":
                color = "#FC9D9A";
                top = 89;
                left = 70;
                break;
            case "Back":
                color = "#F9CDAD";
                top = 28;
                left = 69.25;
                break;
            case "Biceps":
                color = "#C8C8A9";
                top = 31;
                left = 37;
                break;
            case "Cervical":
                color = "#83AF9B";
                top = 15;
                left = 69.25;
                break;
            case "Chest":
                color = "#DBBD5C";
                top = 27;
                left = 23.25;
                break;
            case "Elbow":
                color = "#62E043";
                top = 39;
                left = 15;
                break;
            case "Foot":
                color = "#FFFDB2";
                top = 91;
                left = 24;
                break;
            case "Forearms":
                color = "#B56A29";
                top = 43;
                left = 37;
                break;
            case "Hamstring/Thigh":
                color = "#DD7156";
                top = 60;
                left = 61;
                break;
            case "Hand":
                color = "#B7D145";
                top = 50;
                left = 87;
                break;
            case "Head/Face":
                color = "#E59CDC";
                top = 7;
                left = 23;
                break;
            case "Hip/Pelvis":
                color = "#AC96EA";
                top = 45;
                left = 17;
                break;
            case "Knee":
                color = "#EABF48";
                top = 68;
                left = 20;
                break;
            case "Lower Leg/Calf":
                color = "#F4A4EF";
                top = 75;
                left = 29;
                break;
            case "Neck/Spine":
                color = "#FFA5C9";
                top = 19;
                left = 69.25;
                break;
            case "Quadriceps":
                color = "#3EC98F";
                top = 57;
                left = 32;
                break;
            case "Shin":
                color = "#B2FFE2";
                top = 78;
                left = 18;
                break;
            case "Shoulder/Proximal Humerous":
                color = "#C0B3F9";
                top = 19;
                left = 61;
                break;
            case "Trapezius":
                color = "#55E85D";
                top = 20;
                left = 73;
                break;
            case "Triceps":
                color = "#0BEFBE";
                top = 31;
                left = 82;
                break;
            case "Wrist":
                color = "#9DB2E8";
                top = 49;
                left = 9;
                break;
            default:
                color = "#10B23B";
                top = 88;
                left = 47;
                break;
        }

        return isLegend
            ? {
                  backgroundColor: color
              }
            : {
                  backgroundColor: color,
                  left: left + "%",
                  top: top + "%"
              };
    };

    const getLegend = () => {
        let injuryLocations = new Map();

        props.injuries.forEach(i => {
            if (injuryLocations.has(i.locationOnBody)) {
                injuryLocations.set(i.locationOnBody, injuryLocations.get(i.locationOnBody) + 1);
            } else {
                injuryLocations.set(i.locationOnBody, 1);
            }
        });

        return Array.from(injuryLocations.keys()).map(i => (
            <div className={classes.legendItem}>
                <div className={classes.dotLegend} style={getStyles(i, true)} />
                <Typography variant="body1">
                    {i +
                        " (" +
                        injuryLocations.get(i) +
                        " / " +
                        (injuryLocations.get(i) / props.injuries.length) * 100 +
                        "%)"}
                </Typography>
            </div>
        ));
    };

    return (
        <div className={classes.root}>
            <Grid container spacing={3} style={{ height: "100%", margin: "0px" }}>
                <Grid item xs={6} className={classes.imageContainer}>
                    <img className={classes.image} src={Body}></img>
                    {props.injuries.map(i => (
                        <div className={classes.dot} style={getStyles(i, false)} />
                    ))}
                </Grid>
                <Grid item xs={6} className={classes.legendContainer}>
                    <Typography variant="h6">Injuries Breakdown</Typography>
                    <Divider light style={{ marginBottom: "8px" }} />
                    {getLegend()}
                </Grid>
            </Grid>
        </div>
    );
}
