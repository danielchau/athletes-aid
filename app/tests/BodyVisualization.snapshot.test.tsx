import * as React from 'react'
import BodyVisualization from '../src/components/BodyVisualization'
import renderer from 'react-test-renderer'

let date: Date = new Date("2019-01-01");
    
const mockInjuryNote = {
    createdBy: "Test createdBy",
    createdOn: date,
    content: "Test content"
}

const mockInjuries = [{
    id: "Test id",
    active: true,
    createdOn: date,
    createdBy: "Test createdBy",
    teamName: "Test teamName",
    athleteName: "Test athleteName",
    injuryDate: date,
    isSportsRelated: true,
    eventType: "Test eventType",
    position: "Test position",
    sideOfBody: "Test sideOfBody",
    locationOnBody: "Test locationOnBody",
    injuryType: "Test injuryType",
    severity: 1,
    status: "Test status",
    mechanism: "Test mechanism",
    injuryDescription: "Test injuryDescription",
    otherNotes: [mockInjuryNote]
}]

it('BodyVisualization renders correctly', () => {
  const tree = renderer.create(<BodyVisualization injuries={mockInjuries} />).toJSON()
  expect(tree).toMatchSnapshot()
})
