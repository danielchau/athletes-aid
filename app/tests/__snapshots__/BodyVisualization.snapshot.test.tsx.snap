// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyVisualization renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    style={
      Object {
        "height": "100%",
        "margin": "0px",
      }
    }
  >
    <div
      className="MuiGrid-root makeStyles-imageContainer-2 MuiGrid-item MuiGrid-grid-xs-6"
    >
      <img
        className="makeStyles-image-3"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="makeStyles-dot-4"
        style={
          Object {
            "backgroundColor": "#10B23B",
            "left": "47%",
            "top": "88%",
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root makeStyles-legendContainer-5 MuiGrid-item MuiGrid-grid-xs-6"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Injuries Breakdown
      </h6>
      <hr
        className="MuiDivider-root MuiDivider-light"
        style={
          Object {
            "marginBottom": "8px",
          }
        }
      />
      <div
        className="makeStyles-legendItemContainer-6"
      >
        <div
          className="makeStyles-legendItem-7"
        >
          <div
            className="makeStyles-dotLegend-8"
            style={
              Object {
                "backgroundColor": "#10B23B",
              }
            }
          />
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Test locationOnBody (1 / 100.00%)
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
